tsl/src/nodes/decompress_chunk/planner.c

```diff
@@ -227,6 +227,9 @@ build_decompression_map(DecompressChunkPath *path, List *scan_tlist, Bitmapset *
 
 		path->decompression_map =
 			lappend_int(path->decompression_map, destination_attno_in_uncompressed_chunk);
+		path->is_segmentby_column =
+			lappend_int(path->is_segmentby_column,
+						compression_info && compression_info->segmentby_column_index != 0);
```

Comment:
I think instead of double-walking the 2 Lists in parallell a new struct should be created with 2 int fields; and store&pass those - it would reduce future confusion this might cause

it would be much clearer and would need less comments/etc (and would not need the assert @jnidzwetzki  is asking for a few lines above)
```
struct compressed_col_info {
  int  destination_attno;
  bool is_segmentby_column;
}
```

Reply:
Unfortunately for `custom_private` we have to stick with data types copyable by postgres `_copy*` functions, and this precludes introducing any structs.